packages/editor/src/components/provider/index.native.js

```diff
@@ -151,12 +166,21 @@ export default compose( [
 			getEditorMode,
 		} = select( 'core/edit-post' );
 
+		const {
+			getBlockIndex,
+			getSelectedBlockClientId,
+		} = select( 'core/block-editor' );
+
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const selectedBlockIndex = getBlockIndex( selectedBlockClientId );
+
 		return {
 			mode: getEditorMode(),
 			isReady: isEditorReady(),
 			blocks: getEditorBlocks(),
 			title: getEditedPostAttribute( 'title' ),
 			getEditedPostContent,
+			selectedBlockIndex: selectedBlockIndex,
```

Comment:
Here you can avoid repeating `selectedBlockIndex`

Reply:
Good catch! Fixed!